framework7/index.d.ts

```diff
@@ -0,0 +1,1092 @@
+	accordionClose(item: HTMLElement | string | Dom7.Dom7);
+	accordionToggle(item: HTMLElement | string | Dom7.Dom7);
+	
+	// SmartSelect
+	smartSelectOpen(smartSelect: HTMLElement | string | Dom7.Dom7): void;
+	smartSelectAddOption(select: HTMLElement | string | Dom7.Dom7, optionHTML: string, index: number): void;
+	
+	// Forms
+	formToJSON(form: HTMLElement | string | Dom7.Dom7): any;
+	formFromJSON(form: HTMLElement | string | Dom7.Dom7, formData: Object): void;
+	formGetData(formId: string): void;
+	formDeleteData(formId: string): void;
+	formStoreData(formId: string, formJSON: Object): void;
+	
+	// Tabs
+	showTab(tab: HTMLElement | string | Dom7.Dom7): boolean;
+	
+	// Swiper
+	swiper(swiperContainer: string | HTMLElement | Dom7.Dom7, parameters?: Swiper.SwiperOptions): Swiper;
+	
+	// Photo Browser
+	photoBrowser(parameters: Framework7.PhotoBrowserOptions): Framework7.PhotoBrowser;
+	
+	// Autocomplete
+	autocomplete(parameters: Framework7.AutocompleteOptions): Framework7.Autocomplete;
+	
+	// Picker
+	picker(parameters: Framework7.PickerOptions): Framework7.Picker;
+	
+	// Calendar/Datepicker
+	calendar(parameters: Framework7.CalendarOptions): Framework7.Calendar;
+	
+	// Pull to Refresh
+	pullToRefreshDone(ptrContent: string | HTMLElement | Dom7.Dom7);
+	pullToRefreshTrigger(ptrContent: string | HTMLElement | Dom7.Dom7);
+	destroyPullToRefresh(ptrContent: string | HTMLElement | Dom7.Dom7);
+	initPullToRefresh(ptrContent: string | HTMLElement | Dom7.Dom7);
+	
+	// Infinite Scroll
+	attachInfiniteScroll(container: string | HTMLElement | Dom7.Dom7);
+	detachInfiniteScroll(container: string | HTMLElement | Dom7.Dom7);
+	
+	// Messages
+	messages(messagesContainer: string | HTMLElement | Dom7.Dom7, parameters: Framework7.MessagesOptions): Framework7.Messages;
+	messagebar(messagebarContainer: string | HTMLElement | Dom7.Dom7, parameters: Framework7.MessageBarOptions): Framework7.MessageBar;
+	
+	// Notifications
+	addNotification(parameters: Framework7.NotificationOptions);
+	closeNotification(notificationElement: string | HTMLElement | Dom7.Dom7);
+	
+	// not documented
+	onPageBeforeInit(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageInit(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageBeforeAnimation(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageAfterAnimation(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageBeforeRemove(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageBack(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+	onPageAfterBack(pageName: string, callback: (page: Framework7.PageData) => void): Framework7.PageCallbackObject;
+}
+
+declare module "Framework7" {
```

Comment:
Since it is lowercase on NPM, I think `"framework7"` is better.
Also, avoid `declare module` and instead prefer `export = Framework7;` on its own, un-nested line. If you want a global `Framework7` available you can use `export as namespace Framework7;` on the line below that.